cmake/linker_script/common/common-rom.cmake

```diff
@@ -7,6 +7,7 @@ zephyr_linker_section_obj_level(SECTION init LEVEL PRE_KERNEL_2)
 zephyr_linker_section_obj_level(SECTION init LEVEL POST_KERNEL)
 zephyr_linker_section_obj_level(SECTION init LEVEL APPLICATION)
 zephyr_linker_section_obj_level(SECTION init LEVEL SMP)
+zephyr_iterable_section(NAME _deferred_init KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
```

Comment:
why the `_` prefix in this case ?
We don't use that elsewhere for the linker generator.

symbols are automatically prefixed with `__` in places like this for referencing in consistent ways:
https://github.com/zephyrproject-rtos/zephyr/blob/d3c9a986ec30e4c1fde5092e73d76f7020ea20ee/cmake/linker/ld/ld_script.cmake#L220

Reply:
`zephyr_iterable_section` creates `_<name>_list_start` and `_<name>_list_end` and those were manually added for deferred devices:

https://github.com/zephyrproject-rtos/zephyr/blob/d3c9a986ec30e4c1fde5092e73d76f7020ea20ee/include/zephyr/linker/common-rom/common-rom-kernel-devices.ld#L21-L23

Reply:
Hmn, the `z_deferred_init` might not match now..